File: integration-test-java/src/main/java/org/seasar/doma/it/entity/Department.java
/*
 * Copyright Doma Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.seasar.doma.it.entity;

import java.util.ArrayList;
import java.util.List;
import org.seasar.doma.*;
import org.seasar.doma.it.domain.Identity;
import org.seasar.doma.it.domain.Location;

@Entity(metamodel = @Metamodel)
public class Department {

  @Id Identity<Department> departmentId;

  Integer departmentNo;

  String departmentName;

  Location<Department> location;

  @Version Integer version;

  @OriginalStates Department originalStates;

  @Association List<Employee> employeeList = new ArrayList<>();

  public Identity<Department> getDepartmentId() {
    return departmentId;
  }

  public void setDepartmentId(Identity<Department> departmentId) {
    this.departmentId = departmentId;
  }

  public Integer getDepartmentNo() {
    return departmentNo;
  }

  public void setDepartmentNo(Integer departmentNo) {
    this.departmentNo = departmentNo;
  }

  public String getDepartmentName() {
    return departmentName;
  }

  public void setDepartmentName(String departmentName) {
    this.departmentName = departmentName;
  }

  public Location<Department> getLocation() {
    return location;
  }

  public void setLocation(Location<Department> location) {
    this.location = location;
  }

  public Integer getVersion() {
    return version;
  }

  public void setVersion(Integer version) {
    this.version = version;
  }

  public List<Employee> getEmployeeList() {
    return employeeList;
  }

  public void setEmployeeList(List<Employee> employeeList) {
    this.employeeList = employeeList;
  }
}

File: integration-test-kotlin/src/main/kotlin/org/seasar/doma/it/criteria/Department.kt
package org.seasar.doma.it.criteria

import org.seasar.doma.Entity
import org.seasar.doma.Id
import org.seasar.doma.Metamodel
import org.seasar.doma.OriginalStates
import org.seasar.doma.Transient
import org.seasar.doma.Version

@Entity(metamodel = Metamodel())
class Department {

    @Id
    var departmentId: Int? = null
    var departmentNo: Int? = null
    var departmentName: String? = null
    var location: String? = null

    @Version
    var version: Int? = null

    @OriginalStates
    var originalStates: Department? = null

    @Transient
    var employeeList: List<Employee> = mutableListOf()
}
